src/components/admin/Settings/PrefixAlerts/EditButton.tsx

```diff
@@ -0,0 +1,33 @@
+import { Button } from '@mui/material';
+import AlertSubscriptionDialog from 'components/admin/Settings/PrefixAlerts/Dialog';
+import useAlertSubscriptionDialog from 'components/admin/Settings/PrefixAlerts/useAlertSubscriptionDialog';
+import { FormattedMessage } from 'react-intl';
+
+interface Props {
+    prefix: string;
+}
+
+function AlertEditButton({ prefix }: Props) {
+    const { open, openGenerateAlertDialog, setOpen, subscriptions } =
+        useAlertSubscriptionDialog(prefix);
+
+    return (
+        <>
+            <Button variant="text" onClick={openGenerateAlertDialog}>
+                <FormattedMessage id="cta.edit" />
+            </Button>
+
+            {subscriptions === null ? null : (
+                <AlertSubscriptionDialog
+                    headerId="admin.alerts.dialog.update.header"
+                    open={open}
+                    setOpen={setOpen}
+                    subscriptions={subscriptions}
```

Comment:
Since we have a hook I do not think we need to be passing this stuff as the Dialog itself could access this through the hook.